src/app/@modal/(...)courses/[slug]/manage/page.tsx

```diff
@@ -0,0 +1,62 @@
+import { notFound, redirect } from 'next/navigation'
+
+// import { slugifyCourseName, urlForCourse } from '@lib/courses'
+import { getCourseInfo } from '@services/archive'
+
+import Modal from '@components/Modal'
+import RenameCourse from '@components/tools/RenameCourse'
+import DeleteCourse from '@components/tools/DeleteCourse'
+import { validateRights } from '@services/tkoUserService'
+
+const parseSlug = (slug: string) => {
+  const parsedSlug = slug.match(/(?<id>\d+)-(?<courseSlug>.*)/)
+  if (!parsedSlug || !parsedSlug.groups) {
+    notFound()
+  }
+
+  const id = parseInt(parsedSlug.groups.id, 10)
+
+  if (isNaN(id)) {
+    notFound()
+  }
+
+  return {
+    id,
+    courseSlug: parsedSlug.groups.courseSlug
+  }
+}
+
+const Page = async ({ params }: any) => {
+  const isRights = await validateRights('rename', 'remove')
+
+  if (!isRights) {
+    redirect('/')
```

Comment:
We should probably have some sort of flash here to indicate what went wrong.

Reply:
Reworked to use `validateUserRights` as NextAuth's upcoming version will not allow sharing same session function with route handlers and server components. When Next.js will have better solution for flash messages in server components only environment, then message is added to only one place.